src/activities/ActivityUsageEntity.js

```diff
@@ -200,12 +200,62 @@ export class ActivityUsageEntity extends Entity {
 		await this._setDate(dateValue, Classes.dates.dueDate, 'dueDate');
 	}
 
+	/**
+	 * @returns {boolean} If F18167-flexible-assignment-dates-ux feature is enabled
+	 */
+	hasActivityAvailabilityDates() {
+		const dateEntity = this._getSubEntityByClass(Classes.availabilityDates.availabilityDates);
+		return dateEntity !== undefined && dateEntity !== null;
+	}
+
 	/**
 	 * @returns {string} Start date of the activity usage
 	 */
 	startDate() {
-		const startDate = this._getDateSubEntity(Classes.dates.startDate);
-		return startDate && startDate.properties.date;
+		const availabilityStartDate = this._getAvailabilityStartDateEntity();
+
+		if (availabilityStartDate && availabilityStartDate.properties && availabilityStartDate.properties.dateTime) {
+			return availabilityStartDate.properties.dateTime.date;
+		}
+
+		const startDate = this._getSubEntityByClass(Classes.dates.startDate);
+
+		if (!startDate || !startDate.properties) {
+			return;
+		}
+
+		return startDate.properties.date;
+	}
+
+	/**
+	 * @returns {Number} Start date type of the activity usage
+	 */
+	startDateType() {
+		const availabilityStartDate = this._getAvailabilityStartDateEntity();
+
+		if (!availabilityStartDate || !availabilityStartDate.properties) {
+			return;
+		}
+
+		return availabilityStartDate.properties.dateType;
+	}
+
+	/**
+	 * @returns {Number} Default start date type of the activity usage
+	 */
+	defaultStartDateType() {
+		const dateEntity = this._getSubEntityByClass(Classes.availabilityDates.availabilityDates);
+
+		if (!dateEntity) {
+			return;
+		}
+
+		const action = dateEntity.getActionByName(Classes.availabilityDates.create);
+		if (!action || !action.fields[1]) {
```

Comment:
Is there a better way to check the specific field on the action? I really really don't like doing it this way....

Reply:
There seems to be a `getFieldByName` method here https://search.d2l.dev/xref/BrightspaceHypermediaComponents/siren-sdk/src/activities/quizzes/timing/QuizTimingEntity.js?r=9bccd289#172

Reply:
Updated it to use `getFieldByName`